kernels/volk/volk_4ic_deinterleave_8i_x2.h

```diff
@@ -0,0 +1,109 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Derek Kozel
+ *
+ */
+
+/*!
+ * \page volk_4ic_deinterleave_8i_x2
+ *
+ * \b Overview
+ *
+ * Deinterleaves the complex 4-bit integer vector into I & Q vector data
+ * and converts them to 8-bit integers.
+ *
+ * <b> Dispatcher Prototype</b>
+ * \code
+ * void volk_4ic_deinterleave_8i_x2(int8_t* iBuffer, int8t_t* qBuffer, const lv_4sc_t complexVector, unsigned int num_points) \endcode
+ *
+ * \b Inputs
+ * \li complexVector: The complex input vector.
+ * \li num_points: The number of complex data values to be deinterleaved.
+ *
+ * \b Outputs
+ * \li iBuffer: The I buffer output data.
+ * \li qBuffer: The Q buffer output data.
+ *
+ * \b Example
+ * \code
+ * int N = 10000;
+ *
+ * volk_4ic_deinterleave_8i_x2();
+ *
+ * volk_free(x);
+ * \endcode
+ *
+ */
+
+#ifndef INCLUDED_volk_4ic_deinterleave_8i_x2_a_H
+#define INCLUDED_volk_4ic_deinterleave_8i_x2_a_H
+
+#include <inttypes.h>
+#include <stdio.h>
+
+#ifdef LV_HAVE_GENERIC
+
+static inline void volk_4ic_deinterleave_8i_x2_generic(int8_t* iBuffer,
+                                                       int8_t* qBuffer,
+                                                       const int8_t* complexVector,
+                                                       unsigned int num_points)
+{
+    const int8_t* complexVectorPtr = (const int8_t*)complexVector;
+    int8_t* iBufferPtr = iBuffer;
+    int8_t* qBufferPtr = qBuffer;
+    for (unsigned int number = 0; number < num_points; number++) {
+       *iBufferPtr = (*complexVectorPtr) >> 4;
+       iBufferPtr++;
+       *qBufferPtr = ((*complexVectorPtr) << 4) >> 4;
+       qBufferPtr++;
```

Comment:
I assume you can just post-increment your output pointers in one line.
But it looks like you don't increment your input pointer at all. 
The generic kernels serves as a reference. All other kernel versions are validated against this one. That's how we run tests.